### PR TITLE
UI, Input: disabled DateTime/Duration (fixes 27970)

### PR DESCRIPTION
**Pull request by nhaagen**
https://mantis.ilias.de/view.php?id=27970

### CODE REVIEW

**Review by klees — CHANGES_REQUESTED** [review]
Hi @nhaagen,

thanks for looking into this. The solution in general looks as if it should work. Thanks for adjusting the examples as well.

There is one thing I like you to change: Currently we are storing the `DefaultRenderer`, passed to `render`, in a property on the `Renderer` object, later on modify it and use it to render a glyph for the Input. This approach, however, might cause problems in the future. There only is one `Renderer` per component (or at least we have to assume this) and calls to `render` could be nested, so one `Renderer` might be used in various positions on the call stack. If `render` modifies the instance, we might end up with other states of the `Renderer` instance then we'd expect. In fact, using a property here is just a backdoor to implicitly pass on the `DefaultRenderer` from `render` down to `renderDateTimeInput`, it is not any state the instance need to keep track of. Please explicitely pass the `DefaultRenderer` down to where you need it instead of using this backdoor.

In general that `Renderer` looks extremely messy, which wasn't introduced in this PR. Maybe we should clean up here some times soon.

Best regards!

**Comment by Amstutz** [comment]
Hi @nhaagen 

Thanks a lot for caring for this. 

At we do not have Testrail cases for this, so there is nothing to be adapted. However, we this be the right time to add them?

@nhaagen You may merge as soon as the point of @klees is taken care of.

File: src/UI/Implementation/Component/Input/Field/Renderer.php [hunk 1]
```diff
@@ -48,6 +47,8 @@ public function render(Component\Component $component, RendererInterface $defaul
          */
         $this->checkComponent($component);
 
+        $this->default_renderer = $default_renderer;
+
```
Comment (klees) [inline src/UI/Implementation/Component/Input/Field/Renderer.php]:
```suggestion
```

**Review by klees — APPROVED** [review]
Hi @nhaagen,

good catch removing that `$DIC` dependency in other locations as well. Thx! Will hit merge in a minute.

Best regards!

**Pull request merged**